### PR TITLE
Add DipDup framework to the Build section

### PR DESCRIPTION
**Pull request by m-kus**
@brennancaldwell please review :)

### CODE REVIEW

File: docs/build/dapp-development/dipdup-framework.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+sidebar_position: 4
+hide_table_of_contents: true
+title: "DipDup Framework"
+hide_title: true
+description: >-
+  Create a selective contract indexer in Python or Golang and explose indexed data via GraphQL
```
Comment (ilanbb1) [inline docs/build/dapp-development/dipdup-framework.md:7]: Please change "explose indexed data..." to "explore indexed data..."

File: docs/build/dapp-development/dipdup-framework.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+## DipDup
+
+[DipDup](https://dipdup.net) is a Python framework for creating backends for Tezos dApps.
+
+#### The problem
+
+A minimal setup for a decentralized application is smart contract + web interface working on top of Tezos RPC. That should be enough for simple cases, however once you need history, aggregated data, time series, complex filtering, a custom backend is required. You can surely try to get around by using indexer APIs, like [TzKT](https://api.tzkt.io) or others, but they will never give you the level of customization you want.
```
Comment (ilanbb1) [inline docs/build/dapp-development/dipdup-framework.md:16]: Please edit "You can surely try to get around by using indexer APIs, like [TzKT](https://api.tzkt.io) or others, but they will never give you the level of customization you want." to "You can surely try to get around by using indexer APIs, like [TzKT](https://api.tzkt.io) or others, but they are not the most feasible solutions for the high levels of customization you might want."

File: docs/build/dapp-development/dipdup-framework.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+#### Selective indexing
+
+Here comes "layer-2" or "selective" indexing:
+1. Only the data relevant to your dApp is fetched during the synchronization;
+2. Your internal data model is dApp specific in oppose to blocks, operations, and other generic entities.
```
Comment (ilanbb1) [inline docs/build/dapp-development/dipdup-framework.md:22]: Please edit "in oppose to..." to "as opposed to..."

File: docs/build/dapp-development/dipdup-framework.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+DipDup framework utilizes this approach and abstracts developer from the indexing workflow letting him focus on the business logic.
```
Comment (ilanbb1) [inline docs/build/dapp-development/dipdup-framework.md:24]: Please edit to "The DipDup framework..."

Comment (ilanbb1) [reply docs/build/dapp-development/dipdup-framework.md:24]: Please edit "abstracts developer from the indexing workflow letting him focus on the business logic" to "abstracts developers from the indexing workflow, letting them focus on the business logic."

**Review by ilanbb1 — COMMENTED** [review]
Hey @m-kus ! I left a few small comments to the PR, happy to review and merge once they've all been edited and happy to answer any questions. Thanks!

**Comment by m-kus** [comment]
Thank you a lot for the proofreading!
Fixed👌🏻

**Review by ilanbb1 — APPROVED** [review]

**Pull request merged**